src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -125,6 +125,14 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
             //     V v = (V)(T)e.Current;
             //     /* node.Body */
             // }
+
+            // The scope of the iteration variable is the foreach statement syntax.
```

Comment:
This is not correct. The scope is the controlled statement only (there is a separate instance of the variable per iteration).